contrib/mpi-proxy-split/mpi-wrappers/mana_p2p_update_logs.c

```diff
@@ -66,16 +66,24 @@ void fill_in_log(struct p2p_log_msg *p2p_log) {
     }
   }
 
-  static off_t request_start = 0;
+  static off_t req_start_offset = 0;
   struct p2p_log_request p2p_request;
   int fd2 = dup(fd_request);
+  if (req_start_offset != 0) {
+    lseek(fd2, req_start_offset, SEEK_SET);
+    req_start_offset = 0;
+  }
   while (1) {
     readall(fd2, &p2p_request, sizeof(p2p_request));
     if (p2p_request.request == p2p_log->request) {
       p2p_log->source = p2p_request.source;
       p2p_log->tag = p2p_request.tag;
       break;
     }
+    // The request sequence is out of order
```

Comment:
Could you add more detail to this comment?  What is an example scenario in which the request sequence can become out of order?
Thanks.

Reply:
I don't know what scenario causes out of order in the VASP test. I think this could happen with the following scenario. Since we save request info in MPI Wait, the request will be out of order in which the log file has request 2 first fallowed by request 1. 
T1: Irecv request 1
T2: Irecv request 2
T3: Wait request 2
T4: Wait request 1